### PR TITLE
blockmotd: Allow overriding via command

### LINKED ISSUE #783: block_motd: add command to override the block
I am usually not interested in the motd, but there are two special cases:
* Checking expiration times for nicks and channels `/motd services.server`.
* Connecting to new network as motd usually lists some channels.

It's of course possible to unload block_motd to check it, but it's not as easy as `/znc *block_motd motd services.` and unloading the module is not possible as not-admin user when the module is loaded globally.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/7181511-block_motd-add-command-to-override-the-block?utm_campaign=plugin&utm_content=tracker%2F1759&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1759&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Zarthus**
Fixes #783

### CODE REVIEW

**Comment by Zarthus** [comment]
@Mikaela: Can you verify if this patch is to your satisfaction?

**Comment by Mikaela** [comment]
Based on what you said at IRC (just outputs motd in server buffer) I think it sounds like what I wanted.

File: modules/block_motd.cpp [hunk 1]
```diff
@@ -19,23 +19,43 @@
...
 	MODCONSTRUCTOR(CBlockMotd) {
+		AddHelpCommand();
+		AddCommand("GetMotd", static_cast<CModCommand::ModCmdFunc>(&CBlockMotd::OverrideCommand), "[<server>]", "Override the block with this command. Can optionally specify which server to query.");
 	}
 
 	virtual ~CBlockMotd() {
 	}
 
+	void OverrideCommand(const CString& sLine) {
+		bOverrideMotdBlock = true;
```
Comment (DarthGandalf) [inline modules/block_motd.cpp]: Double negations are unreadable, may you make it e.g. `bTemporaryAcceptMotd`?

File: modules/block_motd.cpp [hunk 2]
```diff
@@ -19,23 +19,43 @@
...
+				sLine = sLine.Token(0) + " 422 " +
+					sLine.Token(2) + " :MOTD blocked by ZNC";
+			}
+			bOverrideMotdBlock = false;
 		}
 		return CONTINUE;
 	}
+
+private:
+	bool bOverrideMotdBlock = false;
```
Comment (DarthGandalf) [inline modules/block_motd.cpp]: `m_` for members

**Comment by Zarthus** [comment]
@darthgandalf: Sorry, I missed your initial comment about renaming the variable, is this better?

**Pull request merged**